### ISSUE TITLE
Simplify webcontents reactor

### ISSUE THREAD

**Issue by fasterthanlime** [itchio/itch#2031]
Test 1:

  * open https://dr-d-king.itch.io/hear-hear-tennis
  * click twitter profile
  * click Twitter home
  * navigate back
  * navigate back
  * navigate forward
  * navigate forward

Test 2:

  * open new tab
  * type 'twitter.com/fasterthanlime' and press Enter
  * navigate back
  * navigate forward
  * there should only be 2 history entries, `itch://new-tab` and `https://twitter.com/fasterthanlime`
  * open tweet
  * navigate back
  * navigate forward
  * click profile on tweet (not @fasterthanlime's)
  * navigate back
  * you should be back to `https://twitter.com/fasterthanlime`
  * the only 'forward' item should be the other profile

Test 3:

  * open https://itch.io/docs/itch
  * click on a few links
  * back/forward at leisure
  * when at end of history (no forward items), navigate back once, and click another link
  * then back and forward again

Test 4:

  * same as test 3, but with https://itch.io/docs/creators/how-buying-works
  * 'How buying works' is a good test (with hashes)

Test 5:

  * navigate to youtube.com
  * click on video, then on several recommendations
  * back/forward at leisure, it should load normally

Test 6:

  * duckduckgo search (type any two words in address bar, press enter)
  * go to result
  * go back

^ test 6 is currently ( 81121ecd4a5168a69b976a8a6f357990fb94e4da ) broken, because it's ignoring the replaceState duckduckgo does

**Comment by fasterthanlime** [comment]
In case you're wondering, here's the most egregious bug btw:

```
11:22:08.631 INFO (m/reac/web-contents) =========================================
11:22:08.631 INFO (m/reac/web-contents) did-navigate https://twitter.com/
11:22:08.631 DEBUG (m/reac/web-contents) W|    https://dr-d-king.itch.io/hear-hear-tennis
11:22:08.631 DEBUG (m/reac/web-contents) W|    https://twitter.com/dr_d_king
11:22:08.631 DEBUG (m/reac/web-contents) W|  > https://twitter.com/
11:22:08.631 DEBUG (m/reac/web-contents) ---------------------------------
11:22:08.631 DEBUG (m/reac/web-contents) S|   itch://new-tab
11:22:08.631 DEBUG (m/reac/web-contents) S|   https://dr-d-king.itch.io/hear-hear-tennis
11:22:08.631 DEBUG (m/reac/web-contents) S|   https://twitter.com/dr_d_king
11:22:08.632 DEBUG (m/reac/web-contents) S| > https://twitter.com/
11:22:08.632 INFO (m/reac/web-contents)
```

That's test 1, we just navigated to the twitter homepage (3rd history item).

Then when we navigate back...

```
11:22:09.621 INFO (m/reac/web-contents) =================================
11:22:09.621 INFO (m/reac/web-contents) navigation-entry-committed https://twitter.com/
11:22:09.621 DEBUG (m/reac/web-contents) currentIndex 1 inPageIndex -1 inPage true
11:22:09.621 DEBUG (m/reac/web-contents) W|    https://dr-d-king.itch.io/hear-hear-tennis
11:22:09.622 DEBUG (m/reac/web-contents) W|  > https://twitter.com/
11:22:09.622 DEBUG (m/reac/web-contents) W|<   https://twitter.com/
11:22:09.622 DEBUG (m/reac/web-contents) ---------------------------------
11:22:09.622 DEBUG (m/reac/web-contents) S|   itch://new-tab
11:22:09.622 DEBUG (m/reac/web-contents) S|   https://dr-d-king.itch.io/hear-hear-tennis
11:22:09.622 DEBUG (m/reac/web-contents) S| > https://twitter.com/dr_d_king
11:22:09.622 DEBUG (m/reac/web-contents) S|   https://twitter.com/
11:22:09.622 DEBUG (m/reac/web-contents) ==> History stayed the same, offset = -1
11:22:09.622 DEBUG (m/reac/web-contents) Assuming in-history navigation (2 => 1)
11:22:09.623 DEBUG (m/reac/web-contents) Nope, the URLs don't match, nvm
11:22:09.623 DEBUG (m/reac/web-contents) ---------------------------------
11:22:09.623 DEBUG (m/reac/web-contents) S|   itch://new-tab
11:22:09.623 DEBUG (m/reac/web-contents) S|   https://dr-d-king.itch.io/hear-hear-tennis
11:22:09.623 DEBUG (m/reac/web-contents) S| > https://twitter.com/dr_d_king
11:22:09.623 DEBUG (m/reac/web-contents) S|   https://twitter.com/
11:22:09.623 INFO (m/reac/web-contents)
```

See the two https://twitter.com instances in `webContents.history` ?

It persists until the `did-navigate-in-page` event:

```
11:22:09.623 INFO (m/reac/web-contents) =========================================
11:22:09.623 INFO (m/reac/web-contents) did-navigate-in-page https://twitter.com/
11:22:09.623 DEBUG (m/reac/web-contents) W|    https://dr-d-king.itch.io/hear-hear-tennis
11:22:09.624 DEBUG (m/reac/web-contents) W|  > https://twitter.com/
11:22:09.624 DEBUG (m/reac/web-contents) W|    https://twitter.com/
11:22:09.624 DEBUG (m/reac/web-contents) ---------------------------------
11:22:09.624 DEBUG (m/reac/web-contents) S|   itch://new-tab
11:22:09.624 DEBUG (m/reac/web-contents) S|   https://dr-d-king.itch.io/hear-hear-tennis
11:22:09.624 DEBUG (m/reac/web-contents) S| > https://twitter.com/dr_d_king
11:22:09.624 DEBUG (m/reac/web-contents) S|   https://twitter.com/
11:22:09.624 INFO (m/reac/web-contents)
```

Our history (S) is correct. The webContents' history (W) is incorrect.

*then* for some reason (I'm guessing twitter.com does a replaceState?) we get this:

```
11:22:09.637 INFO (m/reac/web-contents) =================================
11:22:09.637 INFO (m/reac/web-contents) navigation-entry-committed https://twitter.com/
11:22:09.638 DEBUG (m/reac/web-contents) currentIndex 1 inPageIndex -1 inPage true
11:22:09.638 DEBUG (m/reac/web-contents) W|    https://dr-d-king.itch.io/hear-hear-tennis
11:22:09.638 DEBUG (m/reac/web-contents) W|< > https://twitter.com/
11:22:09.638 DEBUG (m/reac/web-contents) W|    https://twitter.com/
11:22:09.638 DEBUG (m/reac/web-contents) ---------------------------------
11:22:09.638 DEBUG (m/reac/web-contents) S|   itch://new-tab
11:22:09.638 DEBUG (m/reac/web-contents) S|   https://dr-d-king.itch.io/hear-hear-tennis
11:22:09.638 DEBUG (m/reac/web-contents) S| > https://twitter.com/dr_d_king
11:22:09.638 DEBUG (m/reac/web-contents) S|   https://twitter.com/
11:22:09.638 DEBUG (m/reac/web-contents) ==> History stayed the same, offset = 0
11:22:09.638 DEBUG (m/reac/web-contents) Chrome tells us it's a replace!
11:22:09.639 DEBUG (m/reac/web-contents) ---------------------------------
11:22:09.639 DEBUG (m/reac/web-contents) S|   itch://new-tab
11:22:09.639 DEBUG (m/reac/web-contents) S|   https://dr-d-king.itch.io/hear-hear-tennis
11:22:09.639 DEBUG (m/reac/web-contents) S| > https://twitter.com/
11:22:09.639 DEBUG (m/reac/web-contents) S|   https://twitter.com/
11:22:09.640 INFO (m/reac/web-contents)
```

And now our history is wrong (last four `S` lines).

However, if we try to ignore these, then we'll get issues when using most HTML5 apps that do replaceState - for example, DuckDuckGo does a replaceState when showing results with videos, which will interfere with our safety measure (comparing URLs in the W history with the S history before we use webContents.goToOffset - otherwise we clear the webContent's history and use loadURL).

**Comment by fasterthanlime** [comment]
I'm calling it fixed, since 966064de35030a89e923068d03f7fa3b88917cc8

Guess we'll have to keep an eye on Electron releases, see if we're able to remove workarounds in the future.
